helpers/auth.js

```diff
@@ -0,0 +1,45 @@
+function loginRequired(req, res, next) {
+  if (!req.session.user) return res.status(401).redirect('/auth/login'); // 401 or 403?
+  return next();
+}
+
+function isLoggedIn(req, res, next) {
+  if (req.session.user) {
+    req.flash('success', 'Already logged in!');
+    return res.redirect('/');
+  }
+  return next();
+}
+
+function donorRequired(res, req, next) {
+  if (!req.session.user) return res.redirect('/auth/login');
+
+  return next();
+}
+
+function recipientRequired(res, req, next) {
+  return next();
+}
```

Comment:
Did you miss something here?

Reply:
I will be updating this code in the next PR.